src/collective/documentgenerator/search_replace/utils.py

```diff
@@ -0,0 +1,302 @@
+        result.extend(
+            self.search_XML_pod_zone(input_nodes, filename, 'input field', findexpr)
+        )
+        return result
+
+    def search_XML_pod_zone(self, nodes, filename, node_type, findexpr):
+        text_lines = []
+        node_groups = [self.reach_text_node(node) for node in nodes]
+        i = 1
+        for node_group in node_groups:
+            for node in node_group:
+                text = node.data
+                matches = list(findexpr.finditer(text))
+                if matches:
+                    text_lines.append(
+                        {
+                            'matches': matches,
+                            'XMLnode': node,
+                            'node_number': i,
+                            'node_type': node_type
+                        }
+                    )
+            i = i + 1
+        return text_lines
+
+    def reach_text_node(self, node):
+        def recursive_reach_text_node(node, result):
+            if hasattr(node, '__iter__'):
+                for list_node in node:
+                    recursive_reach_text_node(list_node, result)
+            elif node.nodeType == node.TEXT_NODE:
+                result.append(node)
+            else:
+                recursive_reach_text_node(node.childNodes, result)
+            return result
+        return recursive_reach_text_node(node, [])
+
+    def get_result_display(self, searchresult):
+        to_display = []
+        for result in searchresult:
+            text = result['XMLnode'].data
+            textzone = '%s %i' % (result['node_type'], result['node_number'])
+            for match in result['matches']:
+                start = match.start()
+                end = match.end()
+                display_line = ['', '', '']
+                d_start = 0
+                if start > 100:
+                    d_start = start - 100
+                    display_line[0] = '...'
+                d_end = len(text)
+                if end + 100 < len(text):
+                    d_end = end + 100
+                    display_line[2] = '...'
+                if sys.stdout.isatty():
+                    text_display = '%s\033[93m%s\033[0m%s' % (text[d_start:start], text[start:end], text[end:d_end])
+                else:
+                    text_display = text[d_start:d_end]
+                display_line[1] = text_display
+                final_display_line = ''.join(display_line)
+                if display_line[0] or display_line[2]:
+                    display = "  %s : %s > %s" % (textzone, text_display, final_display_line)
+                else:
+                    display = "  %s : %s" % (textzone, text_display)
+                to_display.append(display)
+        return to_display
+
+
+class SearchAndReplacePODTemplateFiles(SearchPODTemplateFiles):
+    """
+    """
+
+    def __init__(self, find_expr, filenames_expr, replace, target_dir=None, ignorecase=False, recursive=False, silent=True, backup=True):
```

Comment:
Sky is the limit?
Je suis contre la limite à 80 caractères, par contre la limite à 120 caractères je trouve çà qd même utile ;-)